.github/workflows/tests.yml

```
@@ -0,0 +1,262 @@
+    outputs:
+      code-changed: ${{ steps.changes.outputs.code }}
+      tests-changed: ${{ steps.changes.outputs.tests }}
+      should-run-tests: ${{ steps.decide.outputs.should-run }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Need full history for proper change detection
+        
+    - name: Check for changes
+      uses: dorny/paths-filter@v2
```

actionlint (1.7.7)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)